spec/qos-classifier/02-integration_spec.lua

```diff
@@ -0,0 +1,181 @@
+local helpers = require "spec.helpers"
+local cjson = require "cjson"
+
+local PLUGIN_NAME = "qos-classifier"
+
+local config = {
+  termination = { 
+      status_code = 302,
+      header_name= "Location",
+      header_value="https://cred.club"
+  },
+  upstream_header_name = "X-QOS-CLASS",
+  node_count = {
+      http_timeout_in_ms = 15,
+      update_initial_delay_in_sec = 5,
+      initial = 1,
+      update_frequency_in_sec = 1
+  }
+}
+
+local function make_class(green, orange, red)
+  classes = {
+    class_1 = {
+      threshold = green,
+      header_value = "Green"
+    },
+    class_2 = {
+      threshold = orange,
+      header_value = "Orange"
+    },
+    class_3 = {
+      threshold = red,
+      header_value = "Red"
+    }
+  }
+  return classes
+end
+
+local function make_requests_to_upstream(host, start_time)
+  local client
+  while true do
+    client = helpers.proxy_client()
+    assert(client:get("/get", {
+      headers = { Host = host },
+    }))
+    ngx.sleep(0.01)
+    if(ngx.now() - start_time > 1.2) then break end
+  end
+end
+
+for _, strategy in helpers.each_strategy() do
+  describe(PLUGIN_NAME .. ": (integration) [#" .. strategy .. "]", function()
+    local bp, db
+  
+    lazy_setup(function()
+      bp, db = helpers.get_db_utils(strategy, nil, { PLUGIN_NAME })
+      local service = bp.services:insert()
+
+      local route1 = bp.routes:insert {
+        hosts   = {  "test1.com" },
+        service = service,
+      }
+
+      local route2 = bp.routes:insert {
+        hosts   = {  "test2.com" },
+        service = service,
+      }
+
+      local route3 = bp.routes:insert {
+        hosts   = {  "test3.com" },
+        service = service,
+      }
+
+      local route4 = bp.routes:insert {
+        hosts   = {  "test4.com" },
+        service = service,
+      }
+
+      -- Enable Plugin on to check termination
+      config.classes = make_class(1,2,3)
+      assert(bp.plugins:insert{
+        name = PLUGIN_NAME,
+        route = route1,
+        config = config 
+      })
+
+      -- Enable Plugin on to check header_value = Red
+      config.classes = make_class(1,2,1000)
+      assert(bp.plugins:insert{
+        name = PLUGIN_NAME,
+        route = route2,
+        config = config 
+      })
+
+      -- Enable Plugin on to check header_value = Orange
+      config.classes = make_class(1,1000,1001)
+      assert(bp.plugins:insert{
+        name = PLUGIN_NAME,
+        route = route3,
+        config = config 
+      })
+
+      -- Enable Plugin on to check header_value = Green
+      config.classes = make_class(1000,1001,1002)
+      assert(bp.plugins:insert{
+        name = PLUGIN_NAME,
+        route = route4,
+        config = config 
+      })
+
+      -- Start kong
+      assert(helpers.start_kong({
+        database   = strategy,
+        nginx_conf = "/kong-plugin/spec/fixtures/custom_nginx.template",
```

Comment:
why is this custom nginx template required?

Reply:
Custom nginx template is required to add `shared_dict` s to nginx configuration which are required by our plugin. Check [81726e1](https://github.com/CRED-CLUB/kong-plugin-qos-classifier/pull/6/commits/81726e13dcc9e18b7e41f61b78de71606e9ea2a1)